dipper/sources/Ensembl.py

```diff
@@ -230,15 +235,14 @@ def _build_biomart_gene_query(self, taxid):
             return None
         q = etree.Element("Query", query_attributes)
 
-        object_attributes = {
-            "name": ensembl_taxon_to_db_map[taxid],
-            "interface": "default"
-        }
-        d = etree.SubElement(q, "Dataset", object_attributes)
-
         # TODO unused?
-        for i in cols_to_fetch:
-            a = etree.SubElement(d, "Attribute", {"name": i})
```

Comment:
SubElement() appends data directly to an xml element, then returns a copy of the new element.  So even though it appears these are unused, removing this breaks the script.
https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.SubElement

I'll change:

    a = etree.SubElement(d, "Attribute", {"name": i})

to 

    etree.SubElement(d, "Attribute", {"name": i})

Reply:
thanks the fewer false negatives for static checking the better